### ISSUE TITLE
Incompatible with Grails 3.2.M2

### ISSUE THREAD

**Issue by KerchumA222** [grails-plugins/grails-database-migration#81]
When executing `dbmGenerateGormChangelog` I get `Command execution error: Bean named 'sessionFactory' must be of type [org.springframework.beans.factory.FactoryBean], but was actually of type [org.hibernate.internal.SessionFactoryImpl]`

Grails: 3.2.0.M2
database-migration: 2.0.0.RC4 and 2.0.0.BUILD-SNAPSHOT
Sample project: https://github.com/KerchumA222/database-migration-issue

**Comment by Kipriz** [comment]
It looks to be a serious reason for this: Liquibase Hibernate extension is not compatible with Hibernate 5 yet. See this issue https://github.com/liquibase/liquibase-hibernate/issues/96
However, it should be done soon.

**Comment by littleiffel** [comment]
Hi, I tried Grails 3.2.0.RC2 with the DB Migration plugin with Hibernate4 and Hibernate5, for both cases I receive the same error as above. I have tried to reconfigure multiple times, but no luck. As test I retried with grails 3.1.11 and everything worked fine to generate the initial changelog from GORM.

So eventhough there seems to be an issue with Hibernate5 there is also a problem with Hibernate4 for grails 3.2.x

**Comment by ajdevries** [comment]
I tried to update the plugin to grails 3.2.x but having issues with not an actual https://github.com/liquibase/liquibase-hibernate hibernate5 release. And after building this project myself the API of the [HibernateDatabase](https://github.com/liquibase/liquibase-hibernate/blob/master/src/main/java/liquibase/ext/hibernate/database/HibernateDatabase.java) class is changed.

Not enough experienced to see what needs to be done to upgrade this plugin.

**Comment by graemerocher** [comment]
First we need a release of `liquibase-hibernate` to come out that is compatible with at least Hibernate 5, then we can deal with the GORM 6 differences

**Comment by ajdevries** [comment]
Yeah.. The master branch is compatible. But not released to a maven repo.

**Comment by ajdevries** [comment]
He's waiting for more people to test.
- https://github.com/liquibase/liquibase-hibernate/issues/96

I tried to update this plugin but not sure about how to handle the API changes

**Comment by kuceram** [comment]
liquibase-hibernate has been updated and released to maven https://github.com/liquibase/liquibase-hibernate/issues/96...

**Comment by jameskleeh** [comment]
@kuceram @ajdevries @littleiffel @Kipriz @KerchumA222 If you all could update your plugin to 3.0.0.BUILD-SNAPSHOT and give it a try, I would appreciate it! I've tested that command with Grails 3.2.1 and Hibernate 5 with success.

**Comment by littleiffel** [comment]
Thanks I wil try it today

**Comment by kuceram** [comment]
@jameskleeh I've tested creating migration script `dbm-gorm-diff` as well as running the application with migrations. Everything works fine, however I faced same non-breaking issues reported [here](https://github.com/grails-plugins/grails-database-migration/issues/64#issuecomment-256739113):

- I also see `INFO 10/28/16 11:37 AM: liquibase: Can not use class org.grails.plugins.databasemigration.liquibase.GormDatabase as a Liquibase service because it does not have a no-argument constructor` in the log
- I cannot see migration log despite my settings `logger("org.grails.plugins.databasemigration", ALL, ['STDOUT'], false)`

Also I had an issue having this setting in `application.yml`
```
dataSource:
   dbCreate: do-nothing
```
It was working with old version. When I change the value to `none` (what is recommended in [Grails Doc](http://docs.grails.org/latest/guide/conf.html#dataSource)), it started to work.
